core/src/main/java/org/apache/carbondata/core/util/DataTypeUtil.java

```diff
@@ -478,7 +478,12 @@ public static Object getDataDataTypeForNoDictionaryColumn(String dimensionValue,
     } else if (actualDataType == DataTypes.TIMESTAMP) {
       return ByteUtil.toXorBytes((Long) dimensionValue);
     } else if (actualDataType == DataTypes.BINARY) {
-      return (byte[]) dimensionValue;
+      if (dimensionValue instanceof String) {
+        return ((String) dimensionValue).getBytes(
```

Comment:
Need to decode here (base 64 or hex)  and return the decoded bytes ?

Reply:
no need to decode in here, only convert string to byte[]